art/estimators/object_detection/tensorflow_faster_rcnn.py

```diff
@@ -0,0 +1,530 @@
+                                       clipped relative to the image window with conditions y_min <= y_max and
+                                       x_min <= x_max.
+        :param groundtruth_classes_list: A list of 1-D tf.float32 tensors of shape [num_boxes] containing the class
+                                         targets with the zero index which is assumed to map to the first
+                                         non-background class.
+        :param groundtruth_weights_list: A list of 1-D tf.float32 tensors of shape [num_boxes] containing weights for
+                                         groundtruth boxes.
+        :return: A tuple of (predictions, losses, detections):
+
+                    - predictions: a dictionary holding "raw" prediction tensors.
+                    - losses: a dictionary mapping loss keys (`Loss/RPNLoss/localization_loss`,
+                              `Loss/RPNLoss/objectness_loss`, `Loss/BoxClassifierLoss/localization_loss`,
+                              `Loss/BoxClassifierLoss/classification_loss`) to scalar tensors representing
+                              corresponding loss values.
+                    - detections: a dictionary containing final detection results.
+        """
+        if obj_detection_model is None:
+            # If obj_detection_model is None, then we need to have parameters filename and url to download, extract
+            # and load the object detection model
+            if filename is None or url is None:
+                raise ValueError(
+                    "Need input parameters `filename` and `url` to download, "
+                    "extract and load the object detection model."
+                )
+
+            # Download and extract
+            path = get_file(filename=filename, path=ART_DATA_PATH, url=url, extract=True)
+
+            # Load model config
+            pipeline_config = path + '/pipeline.config'
+            configs = config_util.get_configs_from_pipeline_file(pipeline_config)
+            configs['model'].faster_rcnn.second_stage_batch_size = configs[
+                'model'
+            ].faster_rcnn.first_stage_max_proposals
+
+            # Load model
+            obj_detection_model = model_builder.build(
+                model_config=configs['model'],
+                is_training=is_training,
+                add_summaries=False
+            )
+
+        # Provide groundtruth
+        groundtruth_classes_list = [
+            tf.one_hot(
+                groundtruth_class,
+                obj_detection_model.num_classes
+            ) for groundtruth_class in groundtruth_classes_list
+        ]
+
+        obj_detection_model.provide_groundtruth(
+            groundtruth_boxes_list=groundtruth_boxes_list,
+            groundtruth_classes_list=groundtruth_classes_list,
+            groundtruth_weights_list=groundtruth_weights_list
+        )
+
+        # Create model pipeline
+        images *= 255.0
```

Comment:
Should we check the `clip_values` to see if `images` is in the [0, 1] range?

Reply:
We checked in the constructor.